src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -89,6 +91,39 @@ bool StoredTableNode::is_column_nullable(const ColumnID column_id) const {
   return table->column_is_nullable(column_id);
 }
 
+std::vector<IndexStatistics> StoredTableNode::indexes_statistics() const {
+  DebugAssert(!left_input() && !right_input(), "StoredTableNode must be a leaf");
+
+  const auto table = StorageManager::get().get_table(table_name);
+  auto stored_indexes_statistics = table->indexes_statistics();
+
+  if (_pruned_column_ids.empty()) {
+    return stored_indexes_statistics;
+  }
+
+  const auto column_id_mapping = column_ids_after_pruning(table->column_count(), _pruned_column_ids);
+
+  // update index statistics
+  for (auto stored_index_stats_iter = stored_indexes_statistics.begin();
+       stored_index_stats_iter != stored_indexes_statistics.end();) {
+    auto update_index_statistics = [&]() {
```

Comment:
Why is this written as a lambda?

Reply:
Because I temporarily forgot about `break` and wanted to achieve an early exit.
I changed it to a nested loop.